lib/jsdom/living/nodes/HTMLTableElement-impl.js

```diff
@@ -16,10 +18,60 @@ class HTMLTableElementImpl extends HTMLElementImpl {
     return firstChildWithHTMLLocalName(this, "thead");
   }
 
+  set tHead(value) {
+    if (value !== null && value._localName !== 'thead') {
+      throw new DOMException(DOMException.HIERARCHY_REQUEST_ERR,
+        "Cannot set a non-header element as a table header");
+    }
+
+    const currentHead = this.tHead;
+    if (currentHead !== null) {
+      this.removeChild(currentHead);
+    }
+
+    if (value !== null) {
+      let insertionPoint = null;
```

Comment:
Maybe factor this out into `function firstNonCaptionNonColgroup(table) { }`, or `function tHeadInsertionPoint(table) { }`?

Reply:
Done